### PR TITLE
Existing Solutions section

### PR DESCRIPTION
**Pull request by chhoumann**
Please see the LaTeX comments as a reading guide. I pose a few questions throughout.

### CODE REVIEW

File: Report/sections/1_analysis/existing_solutions.tex [hunk 1]
```diff
@@ -0,0 +1,39 @@
+\section{Similar Solutions}
+% ! See Notion -> Outline. We should have defined an initial problem which we can refer to. Just like I do below.
+This section aims to give an overview of how other technologies attempt to solve a problem similar to ours. We decided to research these similar solutions such that we can draw inspiration from them.
+
+\subsection{Scratch}\label{ScratchSection}
+\begin{quote}
+    ``\emph{A key goal of Scratch is to introduce programming to those with no previous programming experience.}'' - Maloney et al, 2010\cite{maloneyScratchProgrammingLanguage2010}
```
Comment (chhoumann) [inline Report/sections/1_analysis/existing_solutions.tex:7]:
```suggestion
% ! See Notion -> Outline. We should have defined an initial problem which we can refer to. Just like I do below.
The aim of this section to give an overview of how other technologies attempt to solve a problem similar to ours. We decided to research these similar solutions such that we can draw inspiration from them.

\subsection{Scratch}\label{ScratchSection}
\begin{quote}
    ``\emph{A key goal of Scratch is to introduce programming to those with no previous programming experience.}'' - Maloney et al, 2010\cite{maloneyScratchProgrammingLanguage2010}
```

Comment (chhoumann) [inline Report/sections/1_analysis/existing_solutions.tex:7]:
```suggestion
% ! See Notion -> Outline. We should have defined an initial problem which we can refer to. Just like I do below.
This section aims to give asdasdasdan overview of how other technologies attempt to solve a problem similar to ours. We decided to research these similar solutions such that we can draw inspiration from them.

\subsection{Scratch}\label{ScratchSection}
\begin{quote}
    ``\emph{A key goal of Scratch is to introduce programming to those with no previous programming experience.}'' - Maloney et al, 2010\cite{maloneyScratchProgrammingLanguage2010}
```

File: Report/sections/1_analysis/existing_solutions.tex [hunk 2]
```diff
@@ -0,0 +1,39 @@
+\section{Similar Solutions}
+% ! See Notion -> Outline. We should have defined an initial problem which we can refer to. Just like I do below.
+This section aims to give an overview of how other technologies attempt to solve a problem similar to ours. We decided to research these similar solutions such that we can draw inspiration from them.
```
Comment (chhoumann) [inline Report/sections/1_analysis/existing_solutions.tex:3]: asdasdasd

File: Report/bib/bibliography.bib [hunk 3]
```diff
@@ -52,4 +52,55 @@ @misc{attention_interest
...
+  langid = {american}
+}
+
+
+@online{ScratchImagineProgramStatistics,
+  title = {Scratch - {{Imagine}}, {{Program}}, {{Share}}},
+  url = {https://scratch.mit.edu/statistics/},
+  urldate = {2021-03-07},
+  abstract = {Make games, stories and interactive art with Scratch. (scratch.mit.edu)},
+}
```
Comment (chhoumann) [inline Report/bib/bibliography.bib:72]: - [ ] do this
- [ ] do that
- [ ] osv

File: Report/sections/1_analysis/existing_solutions.tex [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+\subsection{Scratch}\label{ScratchSection}
+\begin{quote}
+    ``\emph{A key goal of Scratch is to introduce programming to those with no previous programming experience.}'' - Maloney et al, 2010\cite{maloneyScratchProgrammingLanguage2010}
+\end{quote}
+
+Scratch\cite{ScratchImagineProgram} is the first programming language that comes to mind - and for good reason. At the time of writing, Scratch has 68 million registered users\cite{ScratchImagineProgramStatistics}. This makes it the most popular solution for a closely adjacent problem to ours. It also indicates that there is a desire to learn how to program through a largely visual and interactive platform.
```
Comment (Ichikamp) [inline Report/sections/1_analysis/existing_solutions.tex:10]: Irrelevant

File: Report/sections/1_analysis/existing_solutions.tex [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+In CodeCombat, users are able to write code in JavaScript or Python\cite{CodeCombatCodingGames}. The user plays through a code-editor, writing lines of code which moves their character. It is similar to accessing their character through an API.
+
+In CodeCombat, you learn about various programming concepts such as syntax, methods, parameters, loops, and variables through the levels in the game\cite{CodeCombatCodingGames}.
+
+\subsection{Conclusion}
+Both \secref{CodeCombatSection} and \secref{ScratchSection} present existing solutions that are designed to introduce users to programming.
+
+They provide different approaches in order to achieve this. CodeCombat provides a representation of programming that is closer to writing code than Scratch, as Scratch provides blocks as an abstraction over textual code.
+Another difference is how users are taught programming. Using Scratch, learning happens implicitly. There are no levels designed to teach you programming concepts - but there are tutorials. The primary focus is on building. CodeCombat, on the other hand, is more explicit in teaching the user programming concepts. Learning is necessary in order to progress through the levels.
```
Comment (Ichikamp) [inline Report/sections/1_analysis/existing_solutions.tex]:
```suggestion
Another difference is how users are taught programming. Using Scratch, learning happens implicitly. There are no levels designed to teach you programming concepts - but there are tutorials. The primary focus is how to build a program. CodeCombat, on the other hand, is more explicit in teaching the user programming concepts. Learning is necessary in order to progress through the levels.
```

**Review by Ichikamp — APPROVED** [review]

**Pull request merged**